### PR TITLE
refactor: lms service testing

### PR DESCRIPTION
**Pull request by muselesscreator**
**TL;DR -** 
Broke the LmsApiService into a few smaller modules, and added docstrings/tests.
even less reliance on the courseId now :-)

JIRA: [AU-27](https://openedx.atlassian.net/browse/AU-27)

**What changed?**

image

**Developer Checklist**
- [x] Test suites passing
- [x] Documentation and test plan updated, if applicable
- [ ] Received code-owner approving review
- [x] Bumped version number [package.json](../package.json)

**Testing Instructions**

[ How should a reviewer test this PR? ]

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

File: src/data/services/lms/api.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+ * fetch.gradebookData(searchText, cohort, track, options)
+ * fetches updated gradebook data based on current filter selections.
+ * Raises an error if assignment grade limits are set, but not assignment.
+ * @param {string} searchText - search text filter
+ * @param {nunber} cohort - selected cohort filter
+ * @param {string} track - selected track filter
+ * @param {object} options - additional optional filter values
+ * @return {Promise} - get response
+ */
+const gradebookData = (searchText, cohort, track, options = {}) => {
```
Comment (nsprenkle) [inline src/data/services/lms/api.js:32]: This is so much nicer 💯

File: src/data/services/lms/urls.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import { configuration } from 'config';
+import { historyRecordLimit } from './constants';
+import { filterQuery, stringifyUrl } from './utils';
+
+const baseUrl = `${configuration.LMS_BASE_URL}`;
+
+const courseId = window.location.pathname.slice(1);
+
+const api = `${baseUrl}/api/`;
+const courses = `${baseUrl}courses/`;
```
Comment (nsprenkle) [inline src/data/services/lms/urls.js:11]: Missing a `/`? Alternatively, I don't think this is used so it could probably be deleted.
```suggestion
const courses = `${baseUrl}/courses/`;
```

**Review by nsprenkle — CHANGES_REQUESTED** [review]
I like where this is going but I'm seeing lots of problems testing it locally. Need to confirm it's not just broken on my system but wanted to raise.

File: src/data/services/lms/urls.js (hunk 2 above)
Comment (muselesscreator) [reply src/data/services/lms/urls.js]: noted and agreed

File: src/components/GradesTab/ScoreViewInput.jsx [hunk 3]
```diff
@@ -24,8 +24,11 @@ export const ScoreViewInput = ({ format, toggleFormat }) => (
     </FormControl>
   </FormGroup>
 );
+ScoreViewInput.defaultProps = {
+  format: 'percent',
+};
 ScoreViewInput.propTypes = {
-  format: PropTypes.string.isRequired,
+  format: PropTypes.string,
```
[the remark this answers is not in the document]
Comment (muselesscreator) [reply src/components/GradesTab/ScoreViewInput.jsx:31]: eh, the issue was that I somehow removed the link to the mapStateToProps field.

File: .eslintrc.js [hunk 4]
```diff
@@ -1,6 +1,13 @@
 const { createConfig } = require('@edx/frontend-build');
 
-const config = createConfig('eslint');
+const config = createConfig('eslint', {
+  rules: {
+    'import/no-named-as-default': 'off',
```
[the remark this answers is not in the document]
Comment (muselesscreator) [reply .eslintrc.js:5]: there is no such thing at edX right now, and these are the import rules that allow much of the unit testing mock-age we are currently relying on.

**Review by nsprenkle — APPROVED** [review]
Looks good 👍

**Review by jnlapierre — APPROVED** [review]
LGTM, loving the consistency of code structure and formatting.
